### ISSUE TITLE
Cannot convert undefined or null to object in 2.5.0

### ISSUE THREAD

**Issue by danielfein** [neo4j/graphql#620]
Hi I upgraded to 2.5.0 and now a custom cypher query is failing. I downgraded to 2.4.0 to rerun the exact same query and it's totally fine, so it's isolated to 2.5.0.

It's a standard query:

```
query explore{
    explore_topics{
      id
      name
      description
      image
    }
    topic_count
  }
```

Built with a fairly simple custom definition:

```
explore_topics: [Topic] @cypher(statement: """
        MATCH (t:Topic)-[:TAGGED]-(p:Post)
        WITH t, COLLECT(p) as posts
        RETURN t
        ORDER BY SIZE(posts) DESC
        LIMIT 10
    """),
```

And if it's helpful here's the full error:

```
{
  "errors": [
    {
      "message": "Cannot convert undefined or null to object",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "explore_topics"
      ],
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": {
          "stacktrace": [
            "TypeError: Cannot convert undefined or null to object",
            "    at Function.entries (<anonymous>)",
            "    at Object.createProjectionAndParams [as default] (/PATH_TO_APP/node_modules/@neo4j/graphql/dist/translate/create-projection-and-params.js:441:49)",
            "    at /PATH_TO_APP/node_modules/@neo4j/graphql/dist/schema/resolvers/cypher.js:66:71",
            "    at Generator.next (<anonymous>)",
            "    at /PATH_TO_APP/node_modules/@neo4j/graphql/dist/schema/resolvers/cypher.js:26:71",
            "    at new Promise (<anonymous>)",
            "    at __awaiter (/PATH_TO_APP/node_modules/@neo4j/graphql/dist/schema/resolvers/cypher.js:22:12)",
            "    at resolve (/PATH_TO_APP/node_modules/@neo4j/graphql/dist/schema/resolvers/cypher.js:44:16)",
            "    at /PATH_TO_APP/node_modules/@graphql-tools/schema/index.cjs.js:40:24",
            "    at processTicksAndRejections (internal/process/task_queues.js:93:5)"
          ]
        }
      }
    }
  ],
  "data": {
    "explore_topics": null
  }
}
```

**Comment by angrykoala** [comment]
Hi @danielfein 
I noticed that the custom Cypher does not look like valid Cypher (variable t is not defined)

I assume the actual definition you are running is:
```gql
explore_topics: [Topic] @cypher(statement: """
        MATCH (t:Topic)-[:TAGGED]-(p:Post)
        WITH t, COLLECT(p) as posts
        RETURN t
        ORDER BY SIZE(posts) DESC
        LIMIT 10
    """)
```
I tried running this with a minimal database with and without Topics and in both cases seemed to work fine. Could you add the relevant type definitions (`Topic` and `Post`) as well as the expected output that you would receive in 2.4.0 please?

**Comment by danielfein** [comment]
Hi @angrykoala sorry about that, in an effort to rush I copied it in weird and the `t:` was actually there in my code, and have just updated my original post. I tried to simplify but have learned the lesson not to do that, as I shared incomplete code.

I did go back to mess with the query to see if I could further isolate it and I was able to see that it's actually the `topic_count` that is messing it all up. The query runs when topic_count is not present in `2.5.0`, and it's defined like this:

```
  topic_count: Int @cypher(statement: """
        MATCH (t:Topic)
        RETURN count(t)
    """),
```

Topic has a very basic definition:

```
type Topic {
    id: ID! @id(autogenerate: true),
    name: String,
    description: String,
    image: String,
    emoji: String,
}
```

Thank you!

**Comment by jaedag** [comment]
Hi yeah I just ran into the same error myself, and I noticed that it happens when you chain two graphql queries together like so 

```
query ($id: ID) {
    users(where: { id: $id } ) {
      id
      name
    }
    businesses(where: { id: $id } ) {
      id
      name
  }
```

Something that used to work flawlessly in the earlier versions of this library. So hoping it gets fixed soon.

**Comment by danielfein** [comment]
> Hi yeah I just ran into the same error myself, and I noticed that it happens when you chain two graphql queries together like so
> 
> ```
> query ($id: ID) {
>     users(where: { id: $id } ) {
>       id
>       name
>     }
>     businesses(where: { id: $id } ) {
>       id
>       name
>   }
> ```
> 
> Something that used to work flawlessly in the earlier versions of this library. So hoping it gets fixed soon.

Yep this is what I noticed as well, it occurs when you have two queries in one

**Comment by angrykoala** [comment]
Hi @danielfein and @jaedag 
Sorry for taking so long to answer, we were able to pinpoint this problem to a race condition happening regarding context when dealing with multiple resolvers in the same query, this issue surfaced due to unrelated changes that caused the race condition to appear in `2.5.0`

It will be patched shortly and should be working fine in `2.5.2`
